### ISSUE TITLE
Add sync mode to async RPC calls

### ISSUE THREAD

**Issue by tromer** [zcash/zcash#1952]
Implement an `--await` argument to `zcash-cli` which, for async operations, does a blocking wait until the operation completes (status `cancel`, `failed` or `success`), and then exits with a corresponding exit code.

For example:
```
zcash-cli --await z_sendmany ...
```

This will make it easier and safer to do z-address transfers from scripts.

**Comment by nathan-at-least** [comment]
Agreed from my experience of scripting a series of transactions, this would be useful. (My script effectively implements this by repeatedly calling `z_getoperationresult` until it completes, sleeping so long as the operation is `executing`.)

Furthermore, I found it useful to also block on the txid getting several confirmations, so it would be nice for the synchronous version to take a 'result-minconf' parameter (not sure if it should be distinct from input-minconf).

**Comment by bladedoyle** [comment]
"Await behaviour shouldn't happen inside the RPC server as it will block one of the rpc threads."

The wait would be very brief though right?  Constructing and broadcasting a transaction should not take so long that holding an rpc thread is a concern.  Any remaining concern could be alleviated by creating the rpc thread pool with a sufficient number of threads, and even making that number tunable to support uncommon use cases.

**Comment by str4d** [comment]
> Constructing and broadcasting a transaction should not take so long that holding an rpc thread is a concern.

For shielded transactions this is not necessarily true. The async RPC methods were introduced when the only shielded protocol was Sprout, which took multiple seconds per input or output; transactions with a few inputs or outputs easily took upwards of 20 seconds. Sapling and Orchard significantly improved on this, but it's still slower than transparent transactions (which don't need proofs, only signatures), and still the case that large shielded transactions can take a noticeably long time to create.
